### PR TITLE
[10.0][MIG] port stock_available_mrp to 10.0

### PR DESCRIPTION
**Pull request by Cedric-Pigeon**
- [x] depends on https://github.com/OCA/stock-logistics-warehouse/pull/279
- [x] depends on https://github.com/OCA/stock-logistics-warehouse/pull/305
- [x] depends on https://github.com/OCA/stock-logistics-warehouse/pull/474

### CODE REVIEW

**Review by lmignon — APPROVED** [review]
LGTM :+1:  (Code review) once #279 is merged

File: stock_available_mrp/models/product_template.py
Comment (lmignon) [inline stock_available_mrp/models/product_template.py]: @Cedric-Pigeon This line should be removed

**Review by congdpt — APPROVED** [review]

File: stock_available_mrp/data/mrp_data.xml
Comment (tafaRU) [inline stock_available_mrp/data/mrp_data.xml]: `noupdate="1"` could be moved at `<odoo>`level and after that remove `<data>`tag.

File: stock_available_mrp/data/mrp_data.xml
Comment (tafaRU) [inline stock_available_mrp/data/mrp_data.xml]: no newline at end of file.

File: stock_available_mrp/views/product_template_view.xml
Comment (tafaRU) [inline stock_available_mrp/views/product_template_view.xml]: please remove `<data>` tag.

**Review by tafaRU — APPROVED** [review]
Just some nitpicking otherwise LGTM

**Comment by MiquelRForgeFlow** [comment]
@Cedric-Pigeon could you rebase this PR, please?

File: stock_available_mrp/models/product_template.py
Comment (atchuthan) [inline stock_available_mrp/models/product_template.py]: the potential quantity needs to be added to immediately_usable_qty
res[template.id]['immediately_usable_qty'] **+**=

File: stock_available_mrp/models/product_template.py
Comment (janarthanandev) [inline stock_available_mrp/models/product_template.py]: It seems you are overwriting the value of immediately_usable_qty to potential_qty. Moreover i think we don't need this method in product.template

**Review by janarthanandev — CHANGES_REQUESTED** [review]

**Comment by gurneyalex** [comment]
travis needs fixing

**Comment by rafaelbn** [comment]
Hi @Cedric-Pigeon this PR is an old one, ready to merge just pending to fix travis :red_circle: Please could you take a look? Thanks!

File: stock_available_mrp/__manifest__.py
Comment (pedrobaeza) [inline stock_available_mrp/__manifest__.py]: Put this in `demo/stock_available_demo.xml`

**Comment by Cedric-Pigeon** [comment]
@tafaRU Done.
@mreficent Rebase done.
@pedrobaeza Done.
@rafaelbn Don understand why travis is red. I rebased the branch, I hope it will solve.

**Review by pedrobaeza — COMMENTED** [review]
I'm thinking that the button should put "Potential to produce", or we will get confused with the rest of the stock_available_* buttons, don't you think?

**Comment by rafaelbn** [comment]
Hi @Cedric-Pigeon :+1: 

Any clue here?

```
2018-05-02 09:15:53,849 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL
2018-05-02 09:15:53,851 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: test_procurement_from_orderpoint_confirmed_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:15:56,024 5310 INFO openerp_test odoo.models.unlink: User #1 deleted stock.move.lots records with IDs: [3]
2018-05-02 09:15:56,110 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL
2018-05-02 09:15:56,113 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: test_procurement_from_orderpoint_draft_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:15:58,381 5310 INFO openerp_test odoo.models.unlink: User #1 deleted stock.move.lots records with IDs: [3]
2018-05-02 09:15:58,466 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL
2018-05-02 09:15:58,469 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: test_procurement_from_orderpoint_sent_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:00,709 5310 INFO openerp_test odoo.models.unlink: User #1 deleted stock.move.lots records with IDs: [3]
2018-05-02 09:16:00,803 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL
2018-05-02 09:16:00,806 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: test_procurement_from_orderpoint_to_approve_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:03,102 5310 INFO openerp_test odoo.models.unlink: User #1 deleted stock.move.lots records with IDs: [3]
2018-05-02 09:16:03,192 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL
2018-05-02 09:16:03,195 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ======================================================================
2018-05-02 09:16:03,196 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL: test_procurement (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:03,196 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: Traceback (most recent call last):
2018-05-02 09:16:03,196 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `   File "/home/travis/build/OCA/stock-logistics-warehouse/purchase_packaging/tests/test_procurement_order.py", line 63, in test_procurement
2018-05-02 09:16:03,196 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `     proc1.purchase_line_id.product_purchase_uom_id)
2018-05-02 09:16:03,196 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ` AssertionError: product.uom(1,) != product.uom()
2018-05-02 09:16:03,196 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ======================================================================
2018-05-02 09:16:03,196 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL: test_procurement_from_orderpoint_confirmed_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: Traceback (most recent call last):
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `   File "/home/travis/build/OCA/stock-logistics-warehouse/purchase_packaging/tests/test_procurement_order.py", line 469, in test_procurement_from_orderpoint_confirmed_po
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `     self.assertTrue(proc.purchase_line_id)
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ` AssertionError: purchase.order.line() is not true
2018-05-02 09:16:03,197 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ======================================================================
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL: test_procurement_from_orderpoint_draft_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: Traceback (most recent call last):
2018-05-02 09:16:03,197 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `   File "/home/travis/build/OCA/stock-logistics-warehouse/purchase_packaging/tests/test_procurement_order.py", line 295, in test_procurement_from_orderpoint_draft_po
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `     self.assertTrue(proc.purchase_line_id)
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ` AssertionError: purchase.order.line() is not true
2018-05-02 09:16:03,198 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ======================================================================
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL: test_procurement_from_orderpoint_sent_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: Traceback (most recent call last):
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `   File "/home/travis/build/OCA/stock-logistics-warehouse/purchase_packaging/tests/test_procurement_order.py", line 351, in test_procurement_from_orderpoint_sent_po
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `     self.assertTrue(proc.purchase_line_id)
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ` AssertionError: purchase.order.line() is not true
2018-05-02 09:16:03,198 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ======================================================================
2018-05-02 09:16:03,198 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAIL: test_procurement_from_orderpoint_to_approve_po (odoo.addons.purchase_packaging.tests.test_procurement_order.TestProcurementOrder)
2018-05-02 09:16:03,199 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: Traceback (most recent call last):
2018-05-02 09:16:03,199 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `   File "/home/travis/build/OCA/stock-logistics-warehouse/purchase_packaging/tests/test_procurement_order.py", line 410, in test_procurement_from_orderpoint_to_approve_po
2018-05-02 09:16:03,199 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: `     self.assertTrue(proc.purchase_line_id)
2018-05-02 09:16:03,199 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: ` AssertionError: purchase.order.line() is not true
2018-05-02 09:16:03,199 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: Ran 5 tests in 12.148s
2018-05-02 09:16:03,199 5310 ERROR openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order: FAILED
2018-05-02 09:16:03,199 5310 INFO openerp_test odoo.addons.purchase_packaging.tests.test_procurement_order:  (failures=5)
2018-05-02 09:16:03,199 5310 INFO openerp_test odoo.modules.module: odoo.addons.purchase_packaging.tests.test_procurement_order tested in 12.15s, 12373 queries
2018-05-02 09:16:03,199 5310 ERROR openerp_test odoo.modules.module: Module purchase_packaging: 5 failures, 0 errors
```

**Comment by rafaelbn** [comment]
Hi @Cedric-Pigeon , I've tested functiionaly and I agree with Pedro in https://github.com/OCA/stock-logistics-warehouse/pull/280#pullrequestreview-116823095 or maybe is a duplicated button?

- I create a "rueda para vender" which is a final product
- I create "rueda sola" and "rueda soporte" which are the material for "rueda para vender"

Note: sorry by Spanish next time english materials...

Check screenshoot, what do you think?

stock_available_mrp_test

**Comment by rafaelbn** [comment]
Hi! This module is being migrated to v11 here https://github.com/OCA/stock-logistics-warehouse/pull/428 maybe you are interested

File: stock_available_mrp/models/product_template.py
Comment (tschanzt) [inline stock_available_mrp/models/product_template.py]: You should add virtual_available to it otherwise your immediately available qty will ignore the stuff that is currently on hand.

**Review by tschanzt — CHANGES_REQUESTED** [review]
Looks good otherwise

**Review by tschanzt — COMMENTED** [review]
additional changes look good but @rafaelbn 's issue still seems unadressed.

**Comment by tschanzt** [comment]
@Cedric-Pigeon Whats the status on this?

**Comment by rousseldenis** [comment]
cc/ @acsone @tschanzt @OCA/logistics-maintainers

**Comment by rafaelbn** [comment]
OMG! :smile: @Cedric-Pigeon all dependencies are merged! please fix travis and let's go merge! I guess your are using this for a while :+1:

**Review by rafaelbn — APPROVED** [review]
:+1: fix travis before merging please! is in :red_circle:

**Comment by Cedric-Pigeon** [comment]
@rafaelbn I won't miss this opportunity. This is my today's goal ;-)

**Comment by Cedric-Pigeon** [comment]
@rafaelbn I fixed some error in travis. One is remaining on purchase_packaging but is not related to this PR. I created a database with the --init done in Travis, installed purchase_packaging but no stock_available* addons and it failed.

**Comment by rafaelbn** [comment]
Well I'm not PSC here @rousseldenis could you merge?

**Review by rafaelbn — APPROVED** [review]
:+1:

**Comment by rousseldenis** [comment]
I'm currently checking that this one would not redify 10 branch, so waiting...

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Fixing

**Comment by rousseldenis** [comment]
Waiting for :
* [x] #542 

As this pull mrp installation, some tests data can't be reused.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**
